src/uniforms/image_unit.rs

```diff
@@ -0,0 +1,335 @@
+    RGBA16UI,
+    /// The image elements have 3 10-bit unsigned integer components and 1 2-bit alpha component
+    RGB10A2UI,
+    /// The image elements are 4-component 8 bit unsigned integer
+    RGBA8UI,
+    /// The image elements are 2-component 32 bit unsigned integer
+    RG32UI,
+    /// The image elements are 2-component 16 bit unsigned integer
+    RG16UI,
+    /// The image elements are 2-component 8 bit unsigned integer
+    RG8UI,
+    /// The image elements are 1-component 32 bit unsigned integer
+    R32UI,
+    /// The image elements are 1-component 16 bit unsigned integer
+    R16UI,
+    /// The image elements are 1-component 8 bit unsigned integer
+    R8UI,
+
+    /// The image elements are 4-component 32 bit signed integer
+    RGBA32I,
+    /// The image elements are 4-component 16 bit signed integer
+    RGBA16I,
+    /// The image elements are 4-component 8 bit signed integer
+    RGBA8I,
+    /// The image elements are 2-component 32 bit signed integer
+    RG32I,
+    /// The image elements are 2-component 16 bit signed integer
+    RG16I,
+    /// The image elements are 2-component 8 bit signed integer
+    RG8I,
+    /// The image elements are 1-component 32 bit signed integer
+    R32I,
+    /// The image elements are 1-component 16 bit signed integer
+    R16I,
+    /// The image elements are 1-component 8 bit signed integer
+    R8I,
+
+    /// The image elements are 4-component 16 bit floating point
+    RGBA16,
+    /// The image elements are 3-component 10 bit floating point with 2 alpha bits
+    RGB10A2,
+    /// The image elements are 4-component 8 bit floating point
+    RGBA8,
+    /// The image elements are 2-component 16 bit floating point
+    RG16,
+    /// The image elements are 2-component 8 bit floating point
+    RG8,
+    /// The image elements are 1-component 16 bit floating point
+    R16,
+    /// The image elements are 1-component 8 bit floating point
+    R8,
+
+    /// The image elements are 4-component 16 bit floating point, normalized to the -1.0 to 1.0 range
+    RGBA16snorm,
+    /// The image elements are 4-component 8 bit floating point, normalized to the -1.0 to 1.0 range
+    RGBA8snorm,
+    /// The image elements are 2-component 16 bit floating point, normalized to the -1.0 to 1.0 range
+    RG16snorm,
+    /// The image elements are 2-component 8 bit floating point, normalized to the -1.0 to 1.0 range
+    RG8snorm,
+    /// The image elements are 1-component 16 bit floating point, normalized to the -1.0 to 1.0 range
+    R16snorm,
+    /// The image elements are 1-component 8 bit floating point, normalized to the -1.0 to 1.0 range
+    R8snorm,
+}
+
+impl ImageUnitFormat {
+    fn get_total_bits(&self) -> usize {
+        match self {
+            ImageUnitFormat::RGBA32F => 4*32,
+            ImageUnitFormat::RGBA16F => 2*32,
+            ImageUnitFormat::RG32F => 2*32,
+            ImageUnitFormat::RG16F => 1*32,
```

Comment:
This should be 2*16. Same result, but different thning. Same applies to the other places this occurs in the match.

Reply:
I wrote it this way to reflect the way it is done in the API spec. I've changed it to match your preference, however.